openqaoa/backends/qpus/qaoa_braket_qpu.py

```diff
@@ -0,0 +1,246 @@
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import os
+import numpy as np
+from time import time
+from typing import Optional, List
+# AWS Braket imports
+
+from braket.circuits import Circuit
+from braket.circuits.gates import H
+from braket.circuits.result_types import Probability
+from braket.circuits.free_parameter import FreeParameter
+from braket.jobs.metrics import log_metric
+
+from ...devices import DeviceAWS
+from ...basebackend import QAOABaseBackendShotBased, QAOABaseBackendCloud, QAOABaseBackendParametric
+from ...qaoa_parameters.baseparams import QAOACircuitParams, QAOAVariationalBaseParams
+from ...utilities import flip_counts
+
+
+class QAOAAWSQPUBackend(QAOABaseBackendParametric, QAOABaseBackendCloud, QAOABaseBackendShotBased):
+    """
+    A QAOA simulator as well as for real QPU using AWS as the backend
+
+    Parameters
+    ----------
+    device: `DeviceAWS`
+        An object of the class ``DeviceAWS`` which contains the credentials
+        for accessing the QPU via cloud and the name of the device.
+    circuit_params: `QAOACircuitParams`
+        An object of the class ``QAOACircuitParams`` which contains information on 
+        circuit construction and depth of the circuit.
+    n_shots: `int`
+        The number of shots to be taken for each circuit.
+    prepend_state: `QuantumCircuit`
+        The state prepended to the circuit.
+    append_state: `QuantumCircuit`
+        The state appended to the circuit.
+    init_hadamard: `bool`
+        Whether to apply a Hadamard gate to the beginning of the 
+        QAOA part of the circuit.
+    cvar_alpha: `float`
+        The value of alpha for the CVaR method.
+    disable_qubit_rewiring: `bool`
+        A boolean that determines whether qubit routing on the provider's end is 
+        used. This is False by default. Not all providers provide this feature.
+    """
+
+    def __init__(self,
+                 circuit_params: QAOACircuitParams,
+                 device: DeviceAWS,
+                 n_shots: int,
+                 prepend_state: Optional[Circuit],
+                 append_state: Optional[Circuit],
+                 init_hadamard: bool,
+                 cvar_alpha: float,
+                 qubit_layout: List[int] = [], 
+                 disable_qubit_rewiring: bool = False):
+
+        QAOABaseBackendShotBased.__init__(self,
+                                          circuit_params,
+                                          n_shots,
+                                          prepend_state,
+                                          append_state,
+                                          init_hadamard,
+                                          cvar_alpha)
+        QAOABaseBackendCloud.__init__(self, device)
+
+        self.qureg = Circuit()
```

Comment:
`qureg` is supposed to be the `(qu)antum (reg)ister` , why is it defined as a circuit?